extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -202,11 +205,67 @@
      * against, and an optional transit context used for key derivation, if applicable.
      * If the signature does not match, or if any other error occurs,
      * the service will throw a {@link VaultVerificationBatchException}
-     * 
-     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#verify-signed-data">verify signed data</a>
+     *
      * @param keyName the key that was used to sign the input
      * @param requests a list of items specifying an input and a signature to match against
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#verify-signed-data">verify signed data</a>
      */
     void verifySignature(String keyName, List<VerificationRequest> requests);
 
+    // --- admin operations
```

Comment:
@vsevel Hi Vincent, did we discuss it earlier and agree that the admin operations should be part of the dedicated admin interface such as [this one](https://github.com/quarkusio/quarkus/blob/master/extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultSystemBackendEngine.java) ? Right, that may not be exactly relevant to the transit key management but if we if do it here, then the same question would apply to other interfaces, example, we hide from the users many low -level/admin operations 

Do you see some practical

Reply:
hello @sberyozkin 
I do not remember, may be ;)
`VaultSystemBackendEngine` maps directly to the [System Backend](https://www.vaultproject.io/api-docs/system). it turns out that all operations on the System Backend may be considered _admin_
that is not the case for many of the other vault endpoints.
I do not mind mixing what could be considered as _admin_ vs _non admin_ to some standards. that is what SB is doing actually in [VaultTransitOperations](https://docs.spring.io/spring-vault/docs/current/api/org/springframework/vault/core/VaultTransitOperations.html).
so I did no think we would need to make this separation. do you? would you rather introduce a `VaultAdminTransitSecretEngine`?

Reply:
Hi Vincent, @vsevel, well, yeah, we spent weeks debating it :-) 

Actually I was thinking it, about my last comment; so you refer to SB. The difference between what we have in Quarkus (what you created) and in SB and likely other providers is that here we have a very user centric interface which is unique enough for Quarkus; the interfaces like [this one](https://docs.spring.io/spring-vault/docs/current/api/org/springframework/vault/core/VaultTransitOperations.html) where whatever is available in a given Vault engine is mixed up in the same interface do not impress me :-), it is just the enumeration of all the operations, not the best approach IMHO.

I don't want to make this PR delayed etc, just a general consideration which I'd consider applying myself.

Re `VaultAdminTransitSecretEngine` - IMHO it would offer the right separation of concerns. I reckon it is fair enough to assume that no-one will for example create a secret key and then use it to sign/encrypt something in context of a single client request (executed in a server context) - as it will require 2 background remote operations - we'd have either some management service where these keys are provisioned or a user service where they are used.

The only concern I'd have about `VaultAdminTransitSecretEngine` is that we'd end up with twice as many interfaces as we have now, if we do it per every other interface. But the separation of the concerns seems more important. These inerfaces can sit in a `admin` sub-package for example and even extend the `simple` ones, ex, `VaultAdminTransitSecretEngine extends VaultTransitSecretEngine`... 

Is it convincing enough :-) ? It is really a proposal to discuss as opposed to a review change request as I agree that it will work in the current PR form too

thanks

Reply:
yes, I guess it is a matter 
 - of sticking to the original api to avoid surprising the users (after all hashicorp did not create 2 interfaces admin/non-admin, and sometimes it is difficult to figure out which is which - is `readSecret` non-admin, and `writeSecret` admin? - today we both have them in `VaultKVSecretEngine` and that feels right), 
 - or creating another abstraction. 

in that case, I was leaning toward the first option, because as a user, if I decided to use the transit secret engine, I would search for a class with that name and expect everything to be there. but again there is no right or wrong. it would be nice to have a third opinion. @gsmet?

no strong feeling on my side if a separation of interfaces means better respecting the quarkus philosophy, just a preference for consistency against the original api.

Reply:
Hi @vsevel

Yeah, they have the endpoint and it is indeed simpler for them to just have a single endpoint dealing with all the operations.

Have a look at [VaultClient](https://github.com/quarkusio/quarkus/blob/master/extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/VaultClient.java). We also have for example a super simple for the users to understand interface,  [VaultKVSecretEngine](https://github.com/quarkusio/quarkus/blob/master/extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultKVSecretEngine.java) - I recall `writeSecret` was added at a user request as there was a concrete use case, etc.

At some point there was only `VaultClient` and I do feel the fact that the KV engine users don't see most of `VaultClient` operations is a good thing for the them :-)

However some interfaces a less `strict` about it, Kubernetes one for ex.

I guess we just need to try to be conservative as to how much of the management operations we can put into an otherwise user centric interfaces.

Perhaps in the case of this specific PR it is OK.

Reply:
hello @sberyozkin 

> Have a look at VaultClient

well I think this one is a bad example ;) for one it is an implementation detail, and we should have refactored it as we were adding more and more methods. and even as an implemation detail, it is too much. I will split at some point.

> I do feel the fact that the KV engine users don't see most of VaultClient operations is a good thing for them :-)

definitely